docs/Using-Primary-Outputs-for-Post-Actions.md

```diff
@@ -0,0 +1,408 @@
+  "sources": [
+    {
+      "rename": {
+        "MyTestProject.csproj": "MyFirstTestProject.csproj"
+      }
+    }
+  ],
+  "symbols": {
+    "firstRename": {
+      "type": "parameter",
+      "dataType": "string",
+      "fileRename": "First"
+    }
+  },
+  "primaryOutputs": [
+    {
+      "path": "MyFirstTestProject.csproj"
+    }
+  ],
+  "postActions": [
+    {
+      "description": "Restore NuGet packages required by this project.",
+      "manualInstructions": [
+        { "text": "Run 'dotnet restore'" }
+      ],
+      "actionId": "210D431B-A78B-4D2F-B762-4ED3E3EA9025",
+      "continueOnError": true,
+      "args": {
+        "files": [ "MyTestProject.csproj" ]
+      }
+    }
+  ]
+}
+```
+Note that the example above contains primary output with name <b>after source based rename is applied</b>. The actual primary output returned by API, contains the final file name after symbol based renames are applied. Post action `files` should contain the original location in template definition.
+Example:
+```
+dotnet new TestAssets.TemplateWithSourceBasedRenames --firstRename Awesome 
+```
+The primary output returned by API is: `MyAwesomeTestProject.csproj`.
+The final location of the files is: `./MyAwesomeTestProject.csproj`.
+
+
+# Different ways of using primary outputs in post actions
+
+Post actions support different ways of choosing the files to process, for more details see the documentation for specific post action.
+
+## Example 1 - apply post action to all primary outputs
+The following example restores all the projects mentioned in primary outputs:
+template files:
+- `./Src/Custom/Path/MyTestProject.csproj`
+- `./Src/Custom/Path/Class.cs`
+`template.json` definition
+```
+{
+  "author": "Test Asset",
+  "classifications": [ "Test Asset" ],
+  "name": "TemplateWithSourceNameAndCustomSourceAndTargetPaths",
+  "generatorVersions": "[1.0.0.0-*)",
+  "groupIdentity": "TestAssets.TemplateWithSourceNameAndCustomSourceAndTargetPaths",
+  "precedence": "100",
+  "identity": "TestAssets.TemplateWithSourceNameAndCustomSourceAndTargetPaths",
+  "shortName": "TestAssets.TemplateWithSourceNameAndCustomSourceAndTargetPaths",
+  "sourceName": "MyTestProject",
+  "sources": [
+    {
+      "source": "./Src/Custom/Path/",
+      "target": "./Target/Output/"
+    }
+  ],
+  "primaryOutputs": [
+    {
+      "path": "Target/Output/MyTestProject.csproj"        
+    }
+  ],
+  "postActions": [
+    {
+      "description": "Restore NuGet packages required by this project.",
+      "manualInstructions": [
+        { "text": "Run 'dotnet restore'" }
+      ],
+      "actionId": "210D431B-A78B-4D2F-B762-4ED3E3EA9025",
+      "continueOnError": true,
+    }
+  ]
+}
+```
+
+## Example 2 - reference source files in post action arguments
+The following example restores individual files specified in `files` property. Defining them in primary outputs is optional.
+template files:
+- `./Src/Custom/Path/MyTestProject.csproj`
+- `./Src/Custom/Path/Class.cs`
```

Comment:
Similar to the example above, would be nice to have 2 projects under template files. One gets restored, the other doesn't.